### PR TITLE
Migration to DD4hep for ME0 db Loader

### PR DESCRIPTION
**Pull request by slomeo**
**PR description:**

This PR (only for ME0) followed what I did for GEM in the PR #32982  . 

**PR validation:**

1) Validation by using "cmsRun CondTools/Geometry/test/me0geometrywriter.py" (with DDD and DD4hep):
by checking printouts for both DD & DD4Hep build methods within the ME0GeometryParsFromDD Class in order to compare some DetId parameters (see below):

//DDD

(0) ME0GeometryParsFromDD - DDD 
(1) DDD, dimension dx1 109.951, dx2 263.652, dy 435.84, dz 147.5
(2) DDD, tran vector 1059.4  -2.37076e-12  5393.5
(3) DDD, rot matrix 2.44929e-16  -1  2.44929e-16 4.89859e-16  3.67394e-16  -1 1  -3.67394e-16  -1.22465e-16
(4) DDD, Chamber DetID 704643078 Name ME0Box
(1) DDD, dimension dx1 109.951, dx2 263.652, dy 435.84, dz 2.9875
(2) DDD, tran vector 1059.4  -2.32539e-12  5270
(3) DDD, rot matrix 2.44929e-16  -1  2.44929e-16 4.89859e-16  3.67394e-16  -1 1  -3.67394e-16  -1.22465e-16
(5) DDD, Layer DetID 704643334 Name ME0L
(1) DDD, dimension dx1 237.625, dx2 263.652, dy 73.803, dz 2.9875
(2) DDD, tran vector 1421.44  -2.4584e-12  5270
(3) DDD, rot matrix 2.44929e-16  -1  2.44929e-16 4.89859e-16  3.67394e-16  -1 1  -3.67394e-16  -1.22465e-16
(6) DDD, Eta Partion DetID 704651526 Name GHA001 nStrips 384 nPads 192

and so on...

//DD4Hep 

(0) ME0GeometryParsFromDD - DD4HEP 
(1) DD4HEP, dimension dx1 109.951, dx2 263.652, dy 435.84, dz 147.5
(2) DD4HEP, tran vector 1059.4  0  5393.5
(3) DD4HEP, rot matrix -6.12323e-17  -1  -6.12323e-17 0  -1.22465e-16  -1 1  1.22465e-16  1.83697e-16
(4) DD4HEP, Chamber DetID 704643078 Name ME0Box
(1) DD4HEP, dimension dx1 109.951, dx2 263.652, dy 435.84, dz 2.9875
(2) DD4HEP, tran vector 1059.4  -1.51244e-14  5270
(3) DD4HEP, rot matrix -6.12323e-17  -1  -6.12323e-17 0  -1.22465e-16  -1 1  1.22465e-16  1.83697e-16
(5) DD4HEP, Layer DetID 704643334 Name ME0L
(1) DD4HEP, dimension dx1 237.625, dx2 263.652, dy 73.803, dz 2.9875
(2) DD4HEP, tran vector 1421.44  2.92124e-14  5270
(3) DD4HEP, rot matrix -6.12323e-17  -1  -6.12323e-17 0  -1.22465e-16  -1 1  1.22465e-16  1.83697e-16
(6) DD4HEP, Eta Partion DetID 704651526 Name GHA001 nStrips 384 nPads 192

and so on..

The DD part of the code has not been touched by this PR (except for the printouts). Added "/ dd4hep::mm;", where it was necessary, in the DD4Hep part of the code.

2) Validation by using a dump script, as I did for the PRs (#32992  #32982  #32781 ), that reads  a local .mb file produced by Payload scripts has been not performed because some problems are present. Please read the FIXME comment in CondTools/Geometry/test/writehelpers/geometryExtended2026_writer.py

This second validation step will be perfomed when all the problems with Payloads will be fixed.

**if this PR is a backport please specify the original PR and why you need to backport that PR:**

nothing special

### CODE REVIEW

**Comment by civanch** [comment]
please test

File: CondTools/Geometry/test/me0geometrywriter.py [hunk 1]
```diff
@@ -14,6 +24,9 @@
 
 process.ME0GeometryWriter = cms.EDAnalyzer("ME0RecoIdealDBLoader")
 
+#process.ME0GeometryWriter = cms.EDAnalyzer("ME0RecoIdealDBLoader",
+#                                           fromDD4Hep = cms.untracked.bool(False))
```
Comment (cvuosalo) [inline CondTools/Geometry/test/me0geometrywriter.py:28]: Shouldn't these lines be uncommented now, and line 25 deleted? The old DD way should work with `fromDD4hep` set to false, right?

File: CondTools/Geometry/test/writehelpers/splitExtended2026Database.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-conddb_import -f sqlite_file:myfile.db -c sqlite_file:GeometryFileExtended2026D41.db -t XMLFILE_Geometry_TagXX_Extended2026D41_mc -i XMLFILE_Geometry_TagXX_Extended2026D41_mc
+conddb_import -f sqlite_file:myfile.db -c sqlite_file:GeometryFileExtended2026D49.db -t XMLFILE_Geometry_TagXX_Extended2026D49_mc -i XMLFILE_Geometry_TagXX_Extended2026D49_mc
```
Comment (cvuosalo) [inline CondTools/Geometry/test/writehelpers/splitExtended2026Database.sh:3]: All the `conddb_import` lines should be replaced with `conddb`, as in `splitExtended2021Database.sh`. `conddb_import` is deprecated.

Comment (slomeo) [reply CondTools/Geometry/test/writehelpers/splitExtended2026Database.sh]: done

File: CondTools/Geometry/test/me0geometrywriter.py (hunk 1 above)
Comment (slomeo) [reply CondTools/Geometry/test/me0geometrywriter.py]: @cvuosalo The old DD way is the default (i.e fromDD4Hep_ = iC.getUntrackedParameter<bool>("fromDD4Hep", false); inside the ME0RecoIdealDBLoader Class) ad I did for GEM. I deleted the comments inside me0geometrywriter.py

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

**Comment by slomeo** [comment]
please, can "db" and "orp" take a look and eventually sign this PR? do you see any problems? Thanks in advance

**Comment by silviodonato** [comment]
kind reminder @ggovi

**Comment by ggovi** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**